src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs

```diff
@@ -458,5 +459,111 @@ void M(int p)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<LocalFunctionStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(24650, "https://github.com/dotnet/roslyn/issues/24650")]
+        public void TestInvalidLocalFunction_BlockAndExpressionBody()
+        {
+            string source = @"
+class C
+{
+    void M(int p)
+    {
+        /*<bind>*/object F() { return new object(); } => null;/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+ILocalFunctionOperation (Symbol: System.Object F()) (OperationKind.LocalFunction, Type: null, IsInvalid) (Syntax: 'object F()  ...  } => null;')
+  Body: 
+    IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid) (Syntax: '{ return new object(); }')
+      IReturnOperation (OperationKind.Return, Type: null, IsInvalid) (Syntax: 'return new object();')
+        ReturnedValue: 
+          IObjectCreationOperation (Constructor: System.Object..ctor()) (OperationKind.ObjectCreation, Type: System.Object, IsInvalid) (Syntax: 'new object()')
+            Arguments(0)
+            Initializer: 
+              null
+  IgnoredBody: 
+    IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid) (Syntax: '=> null')
+      IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: 'null')
+        ReturnedValue: 
+          IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.Object, Constant: null, IsInvalid, IsImplicit) (Syntax: 'null')
+            Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: True, IsUserDefined: False) (MethodSymbol: null)
+            Operand: 
+              ILiteralOperation (OperationKind.Literal, Type: null, Constant: null, IsInvalid) (Syntax: 'null')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // error CS8057: Block bodies and expression bodies cannot both be provided.
+                //         /*<bind>*/object F() { return new object(); } => null;/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_BlockBodyAndExpressionBody, "object F() { return new object(); } => null;").WithLocation(6, 19),
+                // warning CS8321: The local function 'F' is declared but never used
+                //         /*<bind>*/object F() { return new object(); } => null;/*</bind>*/;
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "F").WithArguments("F").WithLocation(6, 26)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<LocalFunctionStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(24650, "https://github.com/dotnet/roslyn/issues/24650")]
```

Comment:
Does this test hit the issue?

Reply:
Yes. As part of fixing this bug, this expression would have returned an IOperation tree that wasn't part of the main tree.

---
In reply to: [178341100](https://github.com/dotnet/roslyn/pull/25782#discussion_r178341100) [](ancestors = 178341100)